src/shared/highlighted-field/use-highlighted-field.js

```diff
@@ -15,51 +22,53 @@ function gatherHighlightedFieldData({ de, coc, dataValueSet }) {
             ...dataValue,
             valueType,
             canHaveLimits,
-            categoryOptionCombo: coc.id,
-            name: de.displayName,
-            code: de.code,
+            name: dataElement.displayName,
+            code: dataElement.code,
             commentOptionSetId: commentOptionSet?.id,
-            description: de.description,
+            description: dataElement.description,
         }
     }
 
     return {
         valueType,
         canHaveLimits,
-        categoryOptionCombo: coc.id,
-        dataElement: de.id,
-        name: de.displayName,
+        categoryOptionCombo: categoryOptionComboId,
+        dataElement: dataElement.id,
+        name: dataElement.displayName,
         lastUpdated: '',
         followup: false,
         comment: null,
         storedBy: null,
         code: null,
         commentOptionSetId: commentOptionSet?.id,
-        description: de.description,
+        description: dataElement.description,
     }
 }
 
 export default function useHighlightedField() {
-    const { data: dataValueSet } = useDataValueSet()
-    const { item } = useHighlightedFieldIdContext()
-    const [currentItem, setHighlightedFieldId] = useState(() => {
-        const { de, coc } = item
-        return gatherHighlightedFieldData({ de, coc, dataValueSet })
-    })
+    const item = useHighlightedFieldStore((state) =>
+        state.getHighlightedField()
+    )
+    const dataValue = useValueStore((state) =>
```

Comment:
This is more a general point rather than just for this PP, but I find myself getting confused around the word `item` its used very interchangable and can mean very different things in different parts of the code. Example here

https://github.com/dhis2/data-entry-app/blob/d2745e7dfd83a62d04ec7a0399a3c2c3a7619a69/src/data-workspace/data-details-sidebar/data-details-sidebar.js#L16-L34

Made we should try not to use the work item in the code in future? its such a generic term its hard to understand what it s mean to be sometimes

Reply:
Yeah, I agree. I renamed it to `highlightedFieldId`.